### ISSUE TITLE
Improve AvoidRecreatingDateTimeFormatter to allow creation for 'variable' date-time formats

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#62]
In the following case, creation of DateTimeFormatter in constructor and assigning to final field can be allowed, as on date formatter is created for this class and later reused.

```java
import org.joda.time.format.DateTimeFormat;
import org.joda.time.format.DateTimeFormatter;

public class Foo {

   private final DateTimeFormatter dateFormatter;
   
   public Foo(String dateFormat) {
         tempDateTimeFormatter = DateTimeFormat.forPattern(dateFormat).withLocale(AbstractGraphCreator.DEFAULT_LOCALE);
   }
     
   protected long dateParser(String value) {
            return dateFormatter.parseDateTime(value).getMillis();
    }

}
```

Also allowed for local variables with a parameter from the method call, and added flags for some undetected lines (the lines with NEW!)

```java
import org.joda.time.format.*;

public class Foo {
    
private static final String DEFAULT_DATETIME_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
    
   public static String format(final long milliseconds, String myFormat, String myTimeZone) {
        final DateTimeFormatter formatter1 = DateTimeFormat.forPattern(myFormat); // good: has variable parameter
        final DateTimeFormatter formatter2 = DateTimeFormat.forPattern(DEFAULT_DATETIME_FORMAT); // bad, NEW!
        final DateTimeFormatter formatter3 = DateTimeFormat.forPattern(DEFAULT_DATETIME_FORMAT).withZoneUTC(); // bad, NEW!
        final DateTimeFormatter formatter4 = DateTimeFormat.forPattern(DEFAULT_DATETIME_FORMAT).withZone(ZoneId.of("UTC")); // bad
        final DateTimeFormatter formatter5 = DateTimeFormat.forPattern(DEFAULT_DATETIME_FORMAT).withZone(myTimeZone); // good: has variable parameter
        return formatter1.print(milliseconds);
    }
}
```

**Comment by jborgers** [comment]
#62 mentioned in 2 lines above should be #63
